Rubberduck.Parsing/Preprocessing/VBAPreprocessor.cs

```diff
@@ -16,17 +21,31 @@ public VBAPreprocessor(double vbaVersion)
             _parser = new VBAPrecompilationParser();
         }
 
-        public void PreprocessTokenStream(string moduleName, CommonTokenStream tokenStream, BaseErrorListener errorListener, CancellationToken token)
+        public void PreprocessTokenStream(IVBProject project, string moduleName, CommonTokenStream tokenStream, BaseErrorListener errorListener, CancellationToken token)
         {
             token.ThrowIfCancellationRequested();
             var symbolTable = new SymbolTable<string, IValue>();
             var tree = _parser.Parse(moduleName, tokenStream, errorListener);
             token.ThrowIfCancellationRequested();
             var stream = tokenStream.TokenSource.InputStream;
-            var evaluator = new VBAPreprocessorVisitor(symbolTable, new VBAPredefinedCompilationConstants(_vbaVersion), stream, tokenStream);
+            var evaluator = new VBAPreprocessorVisitor(symbolTable, new VBAPredefinedCompilationConstants(_vbaVersion), GetUserDefinedCompilationArguments(project), stream, tokenStream);
             var expr = evaluator.Visit(tree);
             var processedTokens = expr.Evaluate(); //This does the actual preprocessing of the token stream as a side effect.
             tokenStream.Reset();
         }
+
+        public Dictionary<string, short> GetUserDefinedCompilationArguments(IVBProject project)
+        {
+            // use the TypeLib API to grab the user defined compilation arguments.  must be obtained on the main thread.
+            var providerInst = UiContextProvider.Instance();
+            var task = (new UiDispatcher(providerInst)).StartTask(delegate () {
```

Comment:
Why new up a a `UiDispatcher` here? Wouldn't it be better to constructor inject an `ICompilationArgumentsProvider` into the preprocessor into which we constructor inject the `UiDispatcher`?